.github/workflows/pull-requests.yml

```diff
@@ -29,7 +31,8 @@ jobs:
 
       - name: Install App and Dependencies
         if: steps.poetry-cache.outputs.cache-hit != 'true'
-        run: python env_setup.py
+        # checks should install the local versions not, what's currently released
+        run: poetry install --extras "env_setup run_tests build_docs"
```

Comment:
I was thinking about this a little more over the weekend and I think the better fix is to figure out a way to replace this ` __commands_to_run.insert(2, f"poetry run pip install jgt_tools[{command}]")` with something more along the lines of a common poetry flow.

IMHO, pip installing in the poetry virtualenv like isn't really conducive to the poetry toolchain, but I'm not entirely familiar with the jgt setup for downstream dependencies using jgt.
I would see this as `poetry add --dev jgt_tools[run_tests,build_docs]` which would add jgt_tools to the pyproject.toml and the extras to the dev dependency as [seen here](https://github.com/dsayling/sphinx_gherkindoc/pull/2/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R27). This would then allow the package using jgt_tools to run whatever `poetry run self-check` or `poetry run ....`. Then any user after the dev dependency is added to the project wouldn't even need to run env_setup.py since they would just run `poetry install` and it would setup the dev dependencies. You can see this working [in this PR](https://github.com/dsayling/sphinx_gherkindoc/pull/2/files#diff-40b14129eaf56def335a07d900ca1d666d9064d2bbaaeea6f2dc16280340b3ff) I opened in my fork for testing. Here is [the action run](https://github.com/dsayling/sphinx_gherkindoc/runs/4585861465?check_suite_focus=true).

Reply:
Interesting.  The jgt_tools stuff was much more @bradsbrown and @brolewis initial refactoring work. I don't have a strong opinion and I have to admit at 11 commits, the other PR was a bit dense (esp. on the last day of holiday break :-) ).
I'm going to defer to whatever Brad/Lewis/Ryan have to say philosophically and look at the coding techniques once that direction has been resolved.

Reply:
Admittedly a lot of our decisions around `jgt_tools` was more about what works than what fits better philosophically within Poetry and so I personally have no problem with this change. I think it's clean and explicit and gets the job done.